src/zodbupdate/update.py

```diff
@@ -146,7 +148,8 @@ def records(self):
                 if next is None:
                     break
         elif (IStorageIteration.providedBy(storage) and
-              not storage.supportsUndo()):
+              (not IStorageUndoable.providedBy(self.storage) or
+               not self.storage.supportsUndo())):
```

Comment:
Why `self.storage` when everything else is checked and done on `storage`?

(AFAICT BlobStorage copies the interfaces and delegates the supportsUndo implementation so it shouldn't matter in practice.)

Reply:
Oops!
My mistake, sorry!!
I pushed a corrected version...  ;)